python/services/github_service.py

```diff
@@ -670,3 +698,184 @@ def remove_user_from_gitub(self, user: str):
+                    f"User {user.login} in team {config.github_team} is ignored")
+                continue
+
+            if self._is_user_inactive(user, inactivity_months, repositories):
+                logging.info(
+                    f"User {user.login} in team {config.github_team} is inactive for {inactivity_months} months")
+
+                if config.remove_users:
+                    self._remove_user(user, config.github_team)
+                    logging.info(
+                        f"User {user.login} removed from team {config.github_team}")
+                    users_removed.append(user)
+                else:
+                    users_to_remove.append(user)
+
+        return users_removed, users_to_remove
+
+    def _load_team_config(self, team_config: dict[str, Any]) -> SelfManagedGitHubTeam:
+        # Ignore the # channel prefix if it exists.
+        slack_channel_name = team_config.get('slack_channel', None)
+        if slack_channel_name is not None and slack_channel_name.startswith('#'):
+            slack_channel_name = slack_channel_name.lstrip('#')
+
+        return SelfManagedGitHubTeam(
+            github_team=team_config['github_team'],
+            remove_users=team_config['remove_from_team'],
+            ignore_users=team_config.get('users_to_ignore', list[str]),
+            ignore_repositories=team_config.get(
+                'repositories_to_ignore', list[str]),
+            slack_channel=slack_channel_name
+        )
+
+    def _message_to_users(self, users_removed: list[NamedUser.NamedUser], users_to_remove: list[NamedUser.NamedUser], team_name: str, inactivity_months: int) -> str:
+        removed_names = "\n".join(
+            [f"- {user.login}" for user in users_removed]) if users_removed else "None"
+        to_remove_names = "\n".join(
+            [f"- {user.login}" for user in users_to_remove]) if users_to_remove else "None"
+
+        message = ""
+        if removed_names or to_remove_names:
+            message = (
+                f"Hi team :wave:,\n\n"
+                f"As part of our ongoing efforts to maintain the integrity of our GitHub teams, we have analysed user activity for the last {inactivity_months} months. Here's the summary:\n\n"
+            )
+            if users_removed:
+                message += (
+                    f":outbox_tray: Users Removed:\n{removed_names}\n"
+                    f"*(These users have been automatically removed from the '{team_name}' team due to inactivity.)*\n\n"
+                )
+            if users_to_remove:
+                message += (
+                    f":eyes: Users Seen but Not Removed:\n{to_remove_names}\n"
+                    f"*(These users were identified as inactive in the '{team_name}' team but were not automatically removed. Please review and take appropriate actions.)*\n\n"
+                )
+            message += (
+                "If you have any questions or concerns, please feel free to reach out.\n\n"
```

Comment:
Would it contact details? maybe #ask-operations-engineering

Reply:
Good idea.